code/audits/audit-service.md

```diff
@@ -0,0 +1,79 @@
+# Audit Service on the Satellite
+
+The purpose of this document is to describe the audit service's implementation.
+
+>An assumption in our storage system is that most storage nodes are reasonably well-behaved, and most data is stored faithfully. As long as that assumption holds, Reed-Solomon is able to detect errors and even correct them, via mechanisms such as the Berlekamp-Welch error correction algorithm [37,65].
+
+Excerpt from [Storj Whitepaper](https://storj.io/storjv3.pdf) Section 4.13, where more information and a higher-level description of the audit service can be found.
+
+This document describes the following:
+- Audit Service Components
+- Overview of an Audit
+- Stripe Selection
+- Verification
+- Containment Mode
+- Reverification
+
+# Components
+
+- Cursor - keeps track of auditing location as while iterating through the metainfoDB (or pointerDB)
+- Verifier - verifies the correctness of a given stripe (the bulk of "auditing" occurs here)
+- Service - coordinates the Cursor and Verifier to run the audit process continuously
+- Reporter - collates audit results and records the stats in the overlay cache
+
+# Overview of an Audit
+
+The audit cursor uses the metainfoDB to select a random stripe from a random remote segment. The audit verifier downloads all erasure shares generated for that stripe. The verifier needs at least the Reed-Solomon minimum required number of pieces (K) in order to verify the stripe's correctness. The verifier uses the Berlekamp-Welch algorithm to determine if any shares have been damaged or altered. If so, the verifier marks the culpable nodes as having failed the audit. If the share has not been altered, then the verifier will record the responsible nodes' audit successes. If the node appears to be offline at the time of downloading, the verifier will record the node as offline.
+
+In a special case, if a node appears to be online but does not send the requested share data to the satellite, then the verifier will mark the node as `contained`, and a pending audit will be entered in the ContainmentDB. The contained node will have three more opportunities (the configurable `MaxReverifyCount`) to produce the correct data, before it's automatically marked as an audit failure.
```

Comment:
> The contained node will have three more opportunities (the configurable `MaxReverifyCount`)

It looks to me that we should mention this in other other way around. Something like _the node will have the `MaxVerifyCount` number of opportunities whose default is 3_  rather than saying 3 and after that there is the configurable parameter.